From 2f6d19f125a74a10301b7cad681eb37840cfcb91 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?M=C3=A5rten=20Nordheim?= <marten.nordheim@qt.io>
Date: Wed, 9 Jun 2021 10:41:51 +0200
Subject: [PATCH 246/254] QHttpSocketEngine: Fix memory leak

The reply needs to be parented or it's not guaranteed to be deleted

Pick-to: 6.2 6.1 5.15
Change-Id: I7cbc31682f80dbbd9fb240fff9e6781cb6ca7b36
Reviewed-by: Edward Welbourne <edward.welbourne@qt.io>
(cherry picked from commit d8077a0973f362376b69564a730c74e5c9366f8f)
---
 src/network/socket/qhttpsocketengine.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/network/socket/qhttpsocketengine.cpp b/src/network/socket/qhttpsocketengine.cpp
index ca7680b71f..6629b7eace 100644
--- a/src/network/socket/qhttpsocketengine.cpp
+++ b/src/network/socket/qhttpsocketengine.cpp
@@ -645,7 +645,7 @@ void QHttpSocketEngine::slotSocketReadNotification()
             d->socket->readAll();
             //We're done with the reply and need to reset it for the next connection
             delete d->reply;
-            d->reply = new QHttpNetworkReply;
+            d->reply = new QHttpNetworkReply(QUrl(), this);
         }
 
         if (priv->phase == QAuthenticatorPrivate::Done)
-- 
2.33.1

